src/bmi_lgar.cxx

```diff
@@ -131,6 +132,32 @@ Update()
 
   assert (state->lgar_bmi_input_params->precipitation_mm_per_h >= 0.0);
   assert(state->lgar_bmi_input_params->PET_mm_per_h >=0.0);
+
+  // adaptive time step is set 
+  if (adaptive_timestep){
+    if ( (state->lgar_bmi_input_params->precipitation_mm_per_h > 0.0) || (state->lgar_mass_balance.volon_timestep_cm > 0.0) ){
+      if (state->lgar_bmi_input_params->precipitation_mm_per_h > 10.0 || (state->lgar_mass_balance.volon_timestep_cm > 0.0) ){
+        subtimestep_h = 1.0/12.0;//case where precip > 1 cm/h, or there is ponded head from the last time step
+        state->lgar_bmi_params.timestep_h = 1.0/12.0;
+      }
+      else {
+        subtimestep_h = 1.0/6.0;//case where there is either nonzero precip or ponded head, but both are less than the threshold that requires the smallest internal time step 
+        state->lgar_bmi_params.timestep_h = 1.0/6.0;
+      }
+    }
+    else {//case where the precip = 0 and there is no ponded head from the last time step 
+      subtimestep_h = 1.0;
+      state->lgar_bmi_params.timestep_h = 1.0;
+    }
+  }
```

Comment:
Ok, sounds good -- I've added your code here and a bit more to avoid hardcoding. Implemented your suggestion of having a user specified minimum time step when adaptive_timestep is true.